src/main/java/com/dsrm/dsrmbackend/services/impl/ReservationServiceImpl.java

```diff
@@ -44,4 +44,14 @@ public Page<Reservation> getReservations(Pageable pageable, String userId ){
         }
         return reservationRepo.findAll(reservationSpecification, pageable);
     }
+
+    @Override
+    public Optional<Reservation> deleteReservation(String id) {
+        Optional<Reservation> reservation = reservationRepo.findById(id);
```

Comment:
we do not have to check if entity should be deleted. We can just delete even if entity does not exists. It will pass.